src/libstd/sys/unix/pipe.rs

```diff
@@ -29,34 +30,33 @@ pub fn anon_pipe() -> io::Result<(AnonPipe, AnonPipe)> {
     // CLOEXEC flag is to use the `pipe2` syscall on Linux. This was added in
     // 2.6.27, however, and because we support 2.6.18 we must detect this
     // support dynamically.
-    if cfg!(target_os = "linux") {
+    static TRY_PIPE2: AtomicBool = AtomicBool::new(cfg!(target_os = "linux"));
```

Comment:
Preexisting: 

`pipe2` is also supported on freebsd, netbsd, dragonflybsd, openbsd. These probably should also appear in that `cfg!` above.